### ISSUE TITLE
64 bit, Error running make when updating ffmpeg git–[/build/ffmpeg-git/ffbuild/common.mak:59: libavformat/libsrt.o] Error 1

### ISSUE THREAD

**Issue by ycwan** [m-ab-s/media-autobuild_suite#1725]
``` bash
23:30:55 ┌ ffmpeg git  .................................. [Updates found]
23:30:55 ├ Until an upstream fix is issued, libaom must be disabled when compiling with libsvtav1....
23:30:55 ├ Until an upstream fix is issued, libopencore-amrwb must be disabled when compiling with libsvtav1....
23:30:55 ├ Changing options to comply to nonfree...
CUDA_PATH environment variable not set or directory does not exist.
23:31:09 ├ Compiling static FFmpeg...
23:31:11 ├ Running uninstall...
23:31:11 ├ Running configure...
23:41:10 ├ Running make...
Likely error (tail of the failed operation logfile):
      |                                                  SRTO_SENDER
src/libavformat/libsrt.c: In function 'libsrt_setup':
src/libavformat/libsrt.c:409:5: warning: 'srt_socket' is deprecated [-Wdeprecated-declarations]
  409 |     fd = srt_socket(cur_ai->ai_family, cur_ai->ai_socktype, 0);
      |     ^~
In file included from src/libavformat/libsrt.c:24:
C:/MinGW64/local64/include/srt/srt.h:735:41: note: declared here
  735 | SRT_ATR_DEPRECATED_PX SRT_API SRTSOCKET srt_socket(int, int, int) SRT_ATR_DEPRECATED;
      |                                         ^~~~~~~~~~
make: *** [/build/ffmpeg-git/ffbuild/common.mak:59: libavformat/libsrt.o] Error 1
make failed. Check C:/MinGW64/build/ffmpeg-git/build-static-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
23:46:34   Creating diagnostics file...

All relevant logs have been anonymously uploaded to https://0x0.st/iJPx.zip
Copy and paste [logs.zip](https://0x0.st/iJPx.zip) in the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.
Try running the build again at a later time.

```

[logs.zip](https://0x0.st/iJPx.zip)
[ab-suite.make.log](https://github.com/m-ab-s/media-autobuild_suite/files/4838008/ab-suite.make.log)

**Comment by 1480c1** [comment]
https://github.com/Haivision/srt/issues/1382

**Comment by hydra3333** [comment]
and https://github.com/Haivision/srt/commit/0e2201aff6b379979cec43fee5e8f162717f6346#commitcomment-40195285

**Comment by Ajaja** [comment]
How to change media-suite_compile.sh to use a previous working libsrt commit?

**Comment by Ajaja** [comment]
I'm a newbie here,  have been using this autobuild script only for a couple of weeks, but already faced a bunch of these "not our bug" issues with broken commits.
Now i use  *_extra.sh scripts with "git checkout working_commit_hash" inside to fix such problems.
Is there a more "official" way to "freeze" libraries?

**Comment by 1480c1** [comment]
no, the purpose of the suite is to use the git master. The purpose of the extra scripts is for doing personal customization such as freezing repos etc.

**Comment by ImportTaste** [comment]
ffmpeg_extra.sh
```bash
#!/bin/bash

# Force to the suite to think the package has updates to recompile.
# Alternatively, you can use "touch recompile" for a similar effect.
touch custom_updated

# Commands to run before running configure
_pre_configure(){
    TICKET=$(curl -sSL "https://trac.ffmpeg.org/ticket/8760?format=csv" | sed -n 's/.*,new,.*/open/p;s/.*,open,.*/open/p')
    [ -n "$TICKET" ] &&
        do_patch "https://gist.githubusercontent.com/ImportTaste/55f61077c9c310b613b16bc51341f31b/raw/0001-Replace-deprecated-libsrt-calls.patch" am ||
        (do_print_progress "ffmpeg issue #8760 appears to be resolved, ffmpeg_extra.sh is likely no longer needed." &
                rm $LOCALBUILDDIR/ffmpeg-git/0001-Replace-deprecated-libsrt-calls.patch)
}
```
I probably could've added a check there to make sure libsrt is even being used, but if you're reading this issue you probably need it.

**Comment by ImportTaste** [comment]
@1480c1 actually, I'm legitimately curious, how _would_ I detect whether or not this patch is needed? Say that the user isn't using anything that requires libsrt, or maybe they built something that used it before but now they're not. I'm thinking it would wrap everything in _pre_configure in its own if check so it wouldn't have to bother invoking curl for the ffmpeg issue.

It would be really good to know for future reference, both in terms of m-ab-s and in terms of my scripts in general.

**Comment by 1480c1** [comment]
For FFmpeg patches, I do not think there is a good way to detect such a thing since I do not know if those tickets get immediately closed or not after someone commits a patch to the FFmpeg repo.

Best I can think of would be to download the patch, try a dry run, and if it succeeded, apply the local download patch, else don't do anything and hope that the patch would have been applied as is to the FFmpeg repo.

If we are certain that the main "symbols" we need to check are the deprecated ones, we could do something like `git grep -q 'SRTO_STRICTENC'` and download the patch and apply based on the error code

**Comment by ImportTaste** [comment]
Oh, I was actually talking about how to detect if m-ab-s is going to compile ffmpeg with anything that needs libsrt. 

That aside, checking the code it's patching for the deprecated call is something I probably should've thought of. It only patches one file, so let's do something like this...

```bash
#!/bin/bash

# Force to the suite to think the package has updates to recompile.
# Alternatively, you can use "touch recompile" for a similar effect.
touch custom_updated

# Commands to run before running configure
_pre_configure(){
    [ "$(sed -nz 's/.*SRTO_STRICTENC.*/1/p' < libavformat/libsrt.c)" == "" ] &&
        (do_print_progress "ffmpeg issue #8760 appears to be resolved, ffmpeg_extra.sh is likely no longer needed." &
                rm $LOCALBUILDDIR/ffmpeg-git/0001-Replace-deprecated-libsrt-calls.patch) ||
        do_patch "https://gist.githubusercontent.com/ImportTaste/55f61077c9c310b613b16bc51341f31b/raw/0001-Replace-deprecated-libsrt-calls.patch" am
}
```

**Comment by 1480c1** [comment]
> Oh, I was actually talking about how to detect if m-ab-s is going to compile ffmpeg with anything that needs libsrt.

from build/media-suite_compile.sh, we only check `enabled libsrt`.

Nice job using sed as a grep along with using the file as stdin instead of just passing the file as an argument.

you might want to change the `rm ..` to `rm -f ..` to make sure it doesn't error out if the patch doesn't exist in the first place (thus triggering the do_patch)

**Comment by ImportTaste** [comment]
> you might want to change the rm .. to rm -f .. to make sure it doesn't error out if the patch doesn't exist in the first place (thus triggering the do_patch)

woops, good catch

> Nice job using sed as a grep along with using the file as stdin instead of just passing the file as an argument.

Thanks, I've heard both techniques are very fast, though passing the file in as stdin is something I don't really understand the performance benefits of.

**Comment by 1480c1** [comment]
> Thanks, I've heard both techniques are very fast, though passing the file in as stdin is something I don't really understand the performance benefits of.

I guess difference between letting bash do fopen and printf vs letting sed fopen etc, idk if there's a benefit to either other than that sed has access to the filename and size etc.

**Comment by ImportTaste** [comment]
Just for shiggles I decided to see if there was a way to do the sed thing with POSIX only, no GNU extensions.
```bash
#!/bin/bash

# Force to the suite to think the package has updates to recompile.
# Alternatively, you can use "touch recompile" for a similar effect.
touch custom_updated

# Commands to run before running configure
_pre_configure(){
    [ "$(sed --posix -n '1h;1!H;${g;/.*SRTO_STRICTENC.*/!d;s//1/p}' < libavformat/libsrt.c)" == "" ] &&
        (do_print_progress "ffmpeg issue #8760 appears to be resolved, ffmpeg_extra.sh is likely no longer needed." &
                rm -f $LOCALBUILDDIR/ffmpeg-git/0001-Replace-deprecated-libsrt-calls.patch) ||
        do_patch "https://gist.githubusercontent.com/ImportTaste/55f61077c9c310b613b16bc51341f31b/raw/0001-Replace-deprecated-libsrt-calls.patch" am
}
```

There's also `sed --posix -n ':a;N;$bb;ba;:b;/.*SRTO_STRICTENC.*/!d;s//1/;p' < libavformat/libsrt.c` which more or less does the same thing, but it's slightly longer of a command and harder to interpret at a glance.

**Comment by NullVSNone** [comment]
I'm trying to apply the patch but it's failing, am I doing something wrong? [logs.zip](https://0x0.st/iyUL.zip) 

```
┌ ffmpeg git  ............................................ [Updates found]
├ Changing options to comply to lgplv3...
├ Compiling static FFmpeg...
├ Running pre configure from ffmpeg_extra.sh...
├ Running configure...
├ Running make...
Likely error (tail of the failed operation logfile):
src/libavformat/libsrt.c:337:50: note: each undeclared identifier is reported only once for each function it appears in
src/libavformat/libsrt.c: In function 'libsrt_setup':
src/libavformat/libsrt.c:401:5: warning: 'srt_socket' is deprecated [-Wdeprecated-declarations]
  401 |     fd = srt_socket(cur_ai->ai_family, cur_ai->ai_socktype, 0);
      |     ^~
In file included from src/libavformat/libsrt.c:24:
C:/MABS/local64/include/srt/srt.h:735:41: note: declared here
  735 | SRT_ATR_DEPRECATED_PX SRT_API SRTSOCKET srt_socket(int, int, int) SRT_ATR_DEPRECATED;
      |                                         ^~~~~~~~~~
make: *** [/build/ffmpeg-git/ffbuild/common.mak:60: libavformat/libsrt.o] Error 1
```

Used https://github.com/m-ab-s/media-autobuild_suite/issues/1725#issuecomment-652664267 and https://github.com/m-ab-s/media-autobuild_suite/issues/1725#issuecomment-652679032 separately.

**Comment by ImportTaste** [comment]
> I'm trying to apply the patch but it's failing, am I doing something wrong? [logs.zip](https://0x0.st/iyUL.zip)
> 
> ```
> ┌ ffmpeg git  ............................................ [Updates found]
> ├ Changing options to comply to lgplv3...
> ├ Compiling static FFmpeg...
> ├ Running pre configure from ffmpeg_extra.sh...
> ├ Running configure...
> ├ Running make...
> Likely error (tail of the failed operation logfile):
> src/libavformat/libsrt.c:337:50: note: each undeclared identifier is reported only once for each function it appears in
> src/libavformat/libsrt.c: In function 'libsrt_setup':
> src/libavformat/libsrt.c:401:5: warning: 'srt_socket' is deprecated [-Wdeprecated-declarations]
>   401 |     fd = srt_socket(cur_ai->ai_family, cur_ai->ai_socktype, 0);
>       |     ^~
> In file included from src/libavformat/libsrt.c:24:
> C:/MABS/local64/include/srt/srt.h:735:41: note: declared here
>   735 | SRT_ATR_DEPRECATED_PX SRT_API SRTSOCKET srt_socket(int, int, int) SRT_ATR_DEPRECATED;
>       |                                         ^~~~~~~~~~
> make: *** [/build/ffmpeg-git/ffbuild/common.mak:60: libavformat/libsrt.o] Error 1
> ```
> 
> Used [#1725 (comment)](https://github.com/m-ab-s/media-autobuild_suite/issues/1725#issuecomment-652664267) and [#1725 (comment)](https://github.com/m-ab-s/media-autobuild_suite/issues/1725#issuecomment-652679032) separately.

Still works fine for me. Try deleting your build\ffmpeg-git directory.

**Comment by 1480c1** [comment]
@ImportTaste @hydra3333 can one of you guys try applying this patch during the FFmpeg stage?

https://gist.github.com/1480c1/e6cac062d5f07306f763a9fc5a7e0056

if it works fine, I will try to see if I can send it to ffmpeg-devel ml, but no guarantee since I haven't done that before.

**Comment by ImportTaste** [comment]
I swapped out my patch with yours and it works fine.

**Comment by 1480c1** [comment]
It seems @mypopydev has already started working on it https://patchwork.ffmpeg.org/project/ffmpeg/patch/1594533783-28695-1-git-send-email-mypopydev@gmail.com/ I will reply to that to say it's working

**Comment by kenichi512** [comment]
I don't understand. How to apply those patches or should I keep waiting until new notice?

**Comment by 1480c1** [comment]
> Just for shiggles I decided to see if there was a way to do the sed thing with POSIX only, no GNU extensions.
> 
> ```shell
> #!/bin/bash
> 
> # Force to the suite to think the package has updates to recompile.
> # Alternatively, you can use "touch recompile" for a similar effect.
> touch custom_updated
> 
> # Commands to run before running configure
> _pre_configure(){
>     [ "$(sed --posix -n '1h;1!H;${g;/.*SRTO_STRICTENC.*/!d;s//1/p}' < libavformat/libsrt.c)" == "" ] &&
>         (do_print_progress "ffmpeg issue #8760 appears to be resolved, ffmpeg_extra.sh is likely no longer needed." ) ||
>         do_patch "https://patchwork.ffmpeg.org/project/ffmpeg/patch/1594533783-28695-1-git-send-email-mypopydev@gmail.com/mbox/" am
> }
> ```
> 
> There's also `sed --posix -n ':a;N;$bb;ba;:b;/.*SRTO_STRICTENC.*/!d;s//1/;p' < libavformat/libsrt.c` which more or less does the same thing, but it's slightly longer of a command and harder to interpret at a glance.

**Comment by kenichi512** [comment]
I mean what file should edit to add that code and what is its directory.

**Comment by psisis** [comment]
> 
> 
> I mean what file should edit to add that code and what is its directory.

I also have this problem, wiki https://github.com/m-ab-s/media-autobuild_suite#custom-patches says you just create `ffmpeg_extra.sh` in `build/` and paste the script quoted above. Or I think alternatively you can manually patch ffmpeg-git with the `git apply` command, this https://gist.github.com/1480c1/e6cac062d5f07306f763a9fc5a7e0056

**Comment by kenichi512** [comment]
Well, I achieved apply the patch but now I have this other error

```
Likely error (tail of the failed operation logfile):
D:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.1.0/../../../../i686-w64-mingw32/bin/ld.exe: D:/MABS/local32/lib/libSPIRV.a(GlslangToSpv.cpp.obj):GlslangToSpv.cpp:(.text$_ZN7glslang5TType12setFieldNameERKNSt7__cxx1112basic_stringIcSt11char_traitsIcENS_14pool_allocatorIcEEEE[__ZN7glslang5TType12setFieldNameERKNSt7__cxx1112basic_stringIcSt11char_traitsIcENS_14pool_allocatorIcEEEE]+0x29): undefined reference to `glslang::GetThreadPoolAllocator()'
D:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.1.0/../../../../i686-w64-mingw32/bin/ld.exe: D:/MABS/local32/lib/libSPIRV.a(GlslangToSpv.cpp.obj):GlslangToSpv.cpp:(.text$_ZN7glslang5TType12setFieldNameERKNSt7__cxx1112basic_stringIcSt11char_traitsIcENS_14pool_allocatorIcEEEE[__ZN7glslang5TType12setFieldNameERKNSt7__cxx1112basic_stringIcSt11char_traitsIcENS_14pool_allocatorIcEEEE]+0x97): undefined reference to `glslang::TPoolAllocator::allocate(unsigned int)'
D:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.1.0/../../../../i686-w64-mingw32/bin/ld.exe: D:/MABS/local32/lib/libSPIRV.a(GlslangToSpv.cpp.obj):GlslangToSpv.cpp:(.text$_ZN7glslang5TTypeC1ERKS0_ib[__ZN7glslang5TTypeC1ERKS0_ib]+0x265): undefined reference to `glslang::GetThreadPoolAllocator()'
D:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.1.0/../../../../i686-w64-mingw32/bin/ld.exe: D:/MABS/local32/lib/libSPIRV.a(GlslangToSpv.cpp.obj):GlslangToSpv.cpp:(.text$_ZN7glslang5TTypeC1ERKS0_ib[__ZN7glslang5TTypeC1ERKS0_ib]+0x273): undefined reference to `glslang::TPoolAllocator::allocate(unsigned int)'
D:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.1.0/../../../../i686-w64-mingw32/bin/ld.exe: D:/MABS/local32/lib/libSPIRV.a(GlslangToSpv.cpp.obj):GlslangToSpv.cpp:(.text$_ZN7glslang5TTypeC1ERKS0_ib[__ZN7glslang5TTypeC1ERKS0_ib]+0x2b5): undefined reference to `glslang::GetThreadPoolAllocator()'
D:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.1.0/../../../../i686-w64-mingw32/bin/ld.exe: D:/MABS/local32/lib/libSPIRV.a(GlslangToSpv.cpp.obj):GlslangToSpv.cpp:(.text$_ZN7glslang5TTypeC1ERKS0_ib[__ZN7glslang5TTypeC1ERKS0_ib]+0x2c3): undefined reference to `glslang::TPoolAllocator::allocate(unsigned int)'
D:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.1.0/../../../../i686-w64-mingw32/bin/ld.exe: D:/MABS/local32/lib/libSPIRV.a(GlslangToSpv.cpp.obj):GlslangToSpv.cpp:(.text$_ZN7glslang5TTypeC1ERKS0_ib[__ZN7glslang5TTypeC1ERKS0_ib]+0x2cd): undefined reference to `glslang::GetThreadPoolAllocator()'
D:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.1.0/../../../../i686-w64-mingw32/bin/ld.exe: D:/MABS/local32/lib/libSPIRV.a(GlslangToSpv.cpp.obj):GlslangToSpv.cpp:(.text$_ZN7glslang5TTypeC1ERKS0_ib[__ZN7glslang5TTypeC1ERKS0_ib]+0x316): undefined reference to `glslang::TPoolAllocator::allocate(unsigned int)'
collect2.exe: error: ld returned 1 exit status
make: *** [/build/ffmpeg-git/Makefile:114: ffmpeg_g.exe] Error 1
```

**Comment by psisis** [comment]
I had a similar problem. Delete ffmpeg-git folder, and change the build option to static in `media-autobuild_suite.ini` (`ffmpegB2=1`)

**Comment by kenichi512** [comment]
I have already tried several times deleting folder but still the same problem.

**Comment by psisis** [comment]
Oh, did you try building a static ffmpeg? Change to `ffmpegB2=1` in `media-autobuild_suite.ini`
Otherwise I truly don't know sorry, I had the exact same compile error but it got fixed somehow

**Comment by kenichi512** [comment]
In fact I have it configured like this, so I think that error must come from elsewhere. I have seen in other threads of same error and possible solutions but none works.

**Comment by psisis** [comment]
> 
> 
> In fact I have it configured like this, so I think that error must come from elsewhere. I have seen in other threads of same error and possible solutions but none works.

The problem is glslang per #1734, assumedly you chose a full ffmpeg build, and you’ll have to comment out `--enable-libglslang` in the ffmpeg options

**Comment by NullVSNone** [comment]
Getting ```Patch could not be applied with `git am`. Continuing without patching.```

@1480c1 This appears to be a common issue?

Tried both https://github.com/m-ab-s/media-autobuild_suite/issues/1725#issuecomment-652679032 and https://github.com/m-ab-s/media-autobuild_suite/issues/1725#issuecomment-652574193

**Comment by 1480c1** [comment]
does it print out which patch and where does it print out?

**Comment by NullVSNone** [comment]
> does it print out which patch and where does it print out?
During ffmpeg: [logs.zip](https://0x0.st/ivko.zip)

**Comment by 1480c1** [comment]
you no longer need ffmpeg_extra.sh for applying the patch since the one in patchwork is now applied by default

I think the ffmpeg_extra.sh is causing a conflict

**Comment by NullVSNone** [comment]
Removed the patch and getting the same issue
```
0001-glslang-add-MachineIndependent.patch
        Patch could not be applied with `git am`. Continuing without patching.
├ Changing options to comply to lgplv3...
├ Compiling static FFmpeg...
├ Running configure...
├ Running make...
Likely error (tail of the failed operation logfile):
src/libavformat/libsrt.c:337:50: note: each undeclared identifier is reported only once for each function it appears in
src/libavformat/libsrt.c: In function 'libsrt_setup':
src/libavformat/libsrt.c:401:5: warning: 'srt_socket' is deprecated [-Wdeprecated-declarations]
  401 |     fd = srt_socket(cur_ai->ai_family, cur_ai->ai_socktype, 0);
      |     ^~
In file included from src/libavformat/libsrt.c:24:
C:/media-autobuild_suite-master/local64/include/srt/srt.h:730:41: note: declared here
  730 | SRT_ATR_DEPRECATED_PX SRT_API SRTSOCKET srt_socket(int, int, int) SRT_ATR_DEPRECATED;
      |                                         ^~~~~~~~~~
make: *** [/build/ffmpeg-git/ffbuild/common.mak:60: libavformat/libsrt.o] Error 1
make failed. Check C:/media-autobuild_suite-master/build/ffmpeg-git/build-static-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...
```
[logs.zip](https://0x0.st/ivk2.zip)
Deleted `build\ffmpeg-git` too just in case

**Comment by 1480c1** [comment]
Ah I see, you are using ffmpeg's n4.1.4 tag.

your specific use case cannot be supported in general since all of the patches following afterward are based on FFmpeg's master branch

**Comment by 1480c1** [comment]
you may need to try to rebase the patches yourself and use them locally instead

**Comment by NullVSNone** [comment]
> you may need to try to rebase the patches yourself and use them locally instead

Would I have to rebase it for every library being used? Is there a general guide for rebasing the patches for ffmpeg?

**Comment by 1480c1** [comment]
Probably not, just the ones that are failing, else you would need to remove them.

to first start off, you should just remove the patches that are failing from the compile script and see if you can compile successfully, there's a chance that you might not need certain patches because that patch only implements functionality or changes necessary from a later commit

**Comment by 1480c1** [comment]
if you still can't compile and you are sure you need the patch, if you have git (assuming you might since it comes with the suite) and you're proficient enough to understand what might be conflicting, you could try cloning the repo and checking out the branch or tag and using `git am -3 patch.patch` and try resolving conflicts, and `git add file` and `git format-patch HEAD~ to regenerate a new patch based not the branch or tag of your choosing.

**Comment by NullVSNone** [comment]
Got it - bit of a pain. Right now is the issue from `0001-glslang-add-MachineIndependent.patch` or `libsrt`? I'd rather just remove than fix - so how would I remove the `glslang-add` patch? I already have `--enable-libglslang` removed from `ffmpeg_options.txt`

**Comment by 1480c1** [comment]
just the glslang one, remove
```bash
do_patch "https://gist.githubusercontent.com/1480c1/18f251a03b7657241c98cc8baf93a223/raw/0001-glslang-add-MachineIndependent.patch" am
```
from the compile script

**Comment by NullVSNone** [comment]
Didn't work: [logs](0x0.st/ivnP.zip). It still looks like the issue is from libsrt (I removed the extra.sh, but not the in-suite patch).

Would it be safe to assume that the best way is to hope Haivision or FFmpeg fix this?

**Comment by 1480c1** [comment]
try removing the `grep -q SRTO_STRICTENC libavformat/libsrt.c &&` line

**Comment by NullVSNone** [comment]
Fixed the new GPAC patch: [logs.zip](https://0x0.st/ivhP.zip)

So this is with removing `glslang` and the `grep` line - it still seems as though the patch is failing? Does this patch not work when freezing ffmpeg?
```
FFmpeg-devel-lavf-srt-fix-build-fail-when-used-the-libsrt-1.4.1.patch
        Patch could not be applied with `git am`. Continuing without patching.
├ Changing options to comply to lgplv3...
├ Compiling static FFmpeg...
├ Running configure...
├ Running make...
Likely error (tail of the failed operation logfile):
src/libavformat/libsrt.c:337:50: note: each undeclared identifier is reported only once for each function it appears in
src/libavformat/libsrt.c: In function 'libsrt_setup':
src/libavformat/libsrt.c:401:5: warning: 'srt_socket' is deprecated [-Wdeprecated-declarations]
  401 |     fd = srt_socket(cur_ai->ai_family, cur_ai->ai_socktype, 0);
      |     ^~
In file included from src/libavformat/libsrt.c:24:
C:/media-autobuild_suite-master/local64/include/srt/srt.h:730:41: note: declared here
  730 | SRT_ATR_DEPRECATED_PX SRT_API SRTSOCKET srt_socket(int, int, int) SRT_ATR_DEPRECATED;
      |                                         ^~~~~~~~~~
make: *** [/build/ffmpeg-git/ffbuild/common.mak:60: libavformat/libsrt.o] Error 1
make failed. Check C:/media-autobuild_suite-master/build/ffmpeg-git/build-static-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...
```

**Comment by kenichi512** [comment]
If it's not one thing it's another, but there doesn't seem to be a single day that there are no fails.

```
Likely error (tail of the failed operation logfile):
make[2]: Entering directory '/build/gpac-git/build-32bit/applications/gpac'
make[2]: Leaving directory '/build/gpac-git/build-32bit/applications/gpac'
make[2]: Entering directory '/build/gpac-git/build-32bit/applications/mp4box'
D:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.1.0/../../../../i686-w64-mingw32/bin/ld.exe: ../../bin/gcc/libgpac_static.a(constants.o):constants.c:(.data+0x0): multiple definition of `itunes_tags'; main.o:main.c:(.bss+0x138): first defined here
collect2.exe: error: ld returned 1 exit status
make[2]: *** [Makefile:68: MP4Box.exe] Error 1
make[2]: Leaving directory '/build/gpac-git/build-32bit/applications/mp4box'
make[1]: *** [Makefile:41: apps] Error 2
make[1]: Leaving directory '/build/gpac-git/build-32bit/applications'
make: *** [Makefile:14: all] Error 2
```

**Comment by NullVSNone** [comment]
@kenichi512 #1713
Might be fixed in latest version - I just removed gpac myself

**Comment by 1480c1** [comment]
for gpac, it seems he may have broken it 7 hours ago https://github.com/gpac/gpac/commit/20e9b1de6934161ce4ada9ec69832dc75a3b9550
